src/lib/y2partitioner/ui_state.rb

```diff
@@ -266,5 +191,136 @@ def create_instance
+      # UI and also to remove useless statuses after deleting a device.
+      #
+      # @see UIState#find_page
+      # @see UIState#clear_statuses_for
+      #
+      # It could hold both, a device id (sid, Integer) or a page label (String).
+      #
+      # @return [Array<Integer, String>]
+      attr_reader :candidate_pages
+
+      # Constructor
+      #
+      # @see UIState#select_page
+      #
+      # @param page_id [String, Integer] Page#id to identify the status and build candidate nodes
+      def initialize(page_id)
+        @page_id = page_id
+        @candidate_pages = build_candidate_pages
+        @selected_devices = { FALLBACK_TAB => nil }
+      end
+
+      # Returns the last selected device for the active tab
+      #
+      # If the node has no tabs a fallback reference will be used. See #selected_devices
+      #
+      # @return [Integer, nil]
+      def selected_device
+        selected_devices[tab]
+      end
+
+      # Stores selected device for the active tab
+      #
+      # If the node has no tabs a fallback reference will be used. See #selected_devices
+      #
+      # @param sid [Integer] the device sid
+      def selected_device=(sid)
+        selected_devices[tab] = sid
+      end
+
+      private
+
+      # The Page id
+      #
+      # @return [String, Integer]
+      attr_reader :page_id
+
+      # A collection to keep the selected devices per tab
+      #
+      # The FALLBACK_TAB key will be used to reference the selected device of a CWM::Page with a
+      # devices table not wrapped within a tab (e.g, Widgets::Pages::System, Widgets:Pages::Disks,
+      # etc)
+      #
+      # @return [Hash{String => Integer}]
+      attr_reader :selected_devices
+
+      # Build the list of candidate pages to go back after opening a device view in the tree
+      #
+      # @return [Array<Integer, String>]
+      def build_candidate_pages
+        if page_id.is_a?(String)
+          [page_id]
+        else
+          device_page_candidates
+        end
+      end
+
+      # List of candidate nodes to go back after opening a device view in the tree
+      #
+      # @return [Array<Integer, String>]
+      def device_page_candidates
+        device = Y2Partitioner::DeviceGraphs.instance.current.find_device(page_id)
+        [device.sid, device_page_parent(device)].compact
+      end
+
+      # @see #device_page_candidates
+      #
+      # @return [Integer, String, nil] nil if there is no parent tree entry
+      def device_page_parent(device)
+        if device.is?(:partition)
+          device.partitionable.sid
+        elsif device.is?(:lvm_lv)
+          device.lvm_vg.sid
+        else
+          device_page_section(device)
+        end
+      end
+
+      # @see #device_page_candidates
+      # @see #device_page_parent
+      #
+      # @return [Integer, String, nil] nil if there is no parent tree entry
+      def device_page_section(device)
```

Comment:
Oh no! We still have a pages dependency!

Reply:
### :skull: